presto-hive/src/main/java/com/facebook/presto/hive/HiveConnector.java

```diff
@@ -154,21 +154,21 @@ public ConnectorTransactionHandle beginTransaction(IsolationLevel isolationLevel
         checkConnectorSupports(READ_UNCOMMITTED, isolationLevel);
         ConnectorTransactionHandle transaction = new HiveTransactionHandle();
         try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(classLoader)) {
-            transactions.put(transaction, metadataFactory.create());
+            transactionManager.put(transaction, metadataFactory.create());
```

Comment:
Is it possible to merge HiveTransactionManager and HiveMetadataFactory since the HiveMetadataFactory is only used here to begin the transaction?